diff --git a/rtl/upstream/main.v b/rtl/upstream/main.v
index 98e7eef..e9c712f 100644
--- a/rtl/upstream/main.v
+++ b/rtl/upstream/main.v
@@ -1,4 +1,15 @@
-module main (
+module main #(
+	parameter reg USE_CX4 = 1'b0,
+	parameter reg USE_SDD1 = 1'b0,
+	parameter reg USE_GSU = 1'b0,
+	parameter reg USE_SA1 = 1'b0,
+	parameter reg USE_DSPn = 1'b0,
+	parameter reg USE_SPC7110 = 1'b0,
+	parameter reg USE_BSX = 1'b0,
+	parameter reg USE_MSU = 1'b0,
+	parameter reg USE_SS = 1'b0,
+	parameter reg USE_SUFAMI = 1'b0
+) (
 	input             RESET_N,
 
 	input             MCLK,
@@ -59,6 +70,7 @@ module main (
 	input       [7:0] CC_DIP,
 
 	input             BLEND,
+	input             DIS_SHORTLINE,
 	input             PAL,
 	output            HIGH_RES,
 	output            V224_MODE,
@@ -129,6 +141,7 @@ module main (
 	input             SS_TOSD,
 	input             SS_LOAD,
 	input       [1:0] SS_SLOT,
+	input      [31:0] SS_GAME_ID,
 	output            SS_AVAIL,
 
 	input      [63:0] SS_DDR_DI,
@@ -139,21 +152,14 @@ module main (
 	output      [7:0] SS_DDR_BE,
 	output            SS_DDR_REQ,
 
+	output            SS_BUSY,
+	output            SS_LOAD_REJECT,
+
 	output     [15:0] AUDIO_L,
 	output     [15:0] AUDIO_R
 );
 
 parameter USE_DLH = 1'b1;
-parameter USE_CX4 = 1'b1;
-parameter USE_SDD1 = 1'b1;
-parameter USE_GSU = 1'b1;
-parameter USE_SA1 = 1'b1;
-parameter USE_DSPn = 1'b1;
-parameter USE_SPC7110 = 1'b1;
-parameter USE_BSX = 1'b1;
-parameter USE_SUFAMI = 1'b1;
-parameter USE_MSU = 1'b1;
-parameter USE_SS = 1'b1;
 
 wire [23:0] CA;
 wire        CPURD_N;
@@ -241,6 +247,7 @@ SNES SNES
 	.sni_joy(SNI_JOY),
 
 	.blend(BLEND),
+	.dis_shortline(DIS_SHORTLINE),
 	.pal(PAL),
 	.high_res(HIGH_RES),
 	.field_out(FIELD),
@@ -274,6 +281,7 @@ SNES SNES
 	.ss_di(SS_DO),
 	.ss_spc_do(SS_SPC_DI),
 	.ss_ppu_do(SS_PPU_DI),
+	.ss_nmi_force(SS_NMI_FORCE),
 
 	.DBG_BG_EN(DBG_BG_EN),
 	.DBG_CPU_EN(DBG_CPU_EN),
@@ -905,8 +913,8 @@ end else
 assign MAP_ACTIVE[6] = 0;
 endgenerate
 
-wire        SS_BUSY;
 wire  [7:0] SS_DO;
+wire        SS_NMI_FORCE;
 wire [23:0] SS_ROM_ADDR;
 
 wire [19:0] SS_EXT_ADDR;
@@ -943,6 +951,7 @@ savestates ss
 	.save_sd(SS_TOSD),
 	.load(SS_LOAD),
 	.slot(SS_SLOT),
+	.ss_game_id(SS_GAME_ID),
 
 	.ram_size(RAM_SIZE),
 	.rom_type(ROM_TYPE),
@@ -1008,7 +1017,9 @@ savestates ss
 
 	.ss_do_ovr(SS_DO_OVR),
 	.ss_rom_ovr(SS_ROM_OVR),
-	.ss_busy(SS_BUSY)
+	.ss_busy(SS_BUSY),
+	.ss_nmi_force(SS_NMI_FORCE),
+	.ss_load_reject(SS_LOAD_REJECT)
 );
 end else begin
 	assign SS_DO = 0;
@@ -1031,6 +1042,8 @@ end else begin
 	assign SS_DO_OVR = 0;
 	assign SS_ROM_OVR = 0;
 	assign SS_BUSY = 0;
+	assign SS_LOAD_REJECT = 0;
+	assign SS_NMI_FORCE = 0;
 end
 endgenerate
 
