diff --git a/rtl/upstream/main.v b/rtl/upstream/main.v
index dba1449..67ec906 100644
--- a/rtl/upstream/main.v
+++ b/rtl/upstream/main.v
@@ -1,4 +1,15 @@
-module main (
+module main #(
+	parameter reg USE_CX4 = 1'b0,
+	parameter reg USE_SDD1 = 1'b0,
+	parameter reg USE_GSU = 1'b0,
+	parameter reg USE_SA1 = 1'b0,
+	parameter reg USE_DSPn = 1'b0,
+	parameter reg USE_SPC7110 = 1'b0,
+	parameter reg USE_BSX = 1'b0,
+	parameter reg USE_MSU = 1'b0,
+	parameter reg USE_SS = 1'b0,
+	parameter reg USE_SUFAMI = 1'b0
+) (
 	input             RESET_N,
 
 	input             MCLK,
@@ -136,21 +147,14 @@ module main (
 	output      [7:0] SS_DDR_BE,
 	output            SS_DDR_REQ,
 
+	output            SS_BUSY,
+	output            SS_LOAD_REJECT,
+
 	output     [15:0] AUDIO_L,
 	output     [15:0] AUDIO_R
 );
 
 parameter USE_DLH = 1'b1;
-parameter USE_CX4 = 1'b1;
-parameter USE_SDD1 = 1'b1;
-parameter USE_GSU = 1'b1;
-parameter USE_SA1 = 1'b1;
-parameter USE_DSPn = 1'b1;
-parameter USE_SPC7110 = 1'b1;
-parameter USE_BSX = 1'b1;
-parameter USE_SUFAMI = 1'b1;
-parameter USE_MSU = 1'b1;
-parameter USE_SS = 1'b1;
 
 wire [23:0] CA;
 wire        CPURD_N;
@@ -270,6 +274,7 @@ SNES SNES
 	.ss_di(SS_DO),
 	.ss_spc_do(SS_SPC_DI),
 	.ss_ppu_do(SS_PPU_DI),
+	.ss_nmi_force(SS_NMI_FORCE),
 
 	.DBG_BG_EN(DBG_BG_EN),
 	.DBG_CPU_EN(DBG_CPU_EN),
@@ -882,8 +887,8 @@ end else
 assign MAP_ACTIVE[6] = 0;
 endgenerate
 
-wire        SS_BUSY;
 wire  [7:0] SS_DO;
+wire        SS_NMI_FORCE;
 wire [23:0] SS_ROM_ADDR;
 
 wire [19:0] SS_EXT_ADDR;
@@ -970,7 +975,9 @@ savestates ss
 
 	.ss_do_ovr(SS_DO_OVR),
 	.ss_rom_ovr(SS_ROM_OVR),
-	.ss_busy(SS_BUSY)
+	.ss_busy(SS_BUSY),
+	.ss_nmi_force(SS_NMI_FORCE),
+	.ss_load_reject(SS_LOAD_REJECT)
 );
 end else begin
 	assign SS_DO = 0;
@@ -991,6 +998,8 @@ end else begin
 	assign SS_DO_OVR = 0;
 	assign SS_ROM_OVR = 0;
 	assign SS_BUSY = 0;
+	assign SS_LOAD_REJECT = 0;
+	assign SS_NMI_FORCE = 0;
 end
 endgenerate
 
