diff --git a/rtl/upstream/PPU.vhd b/rtl/upstream/PPU.vhd
index e83a91f..ef2a766 100644
--- a/rtl/upstream/PPU.vhd
+++ b/rtl/upstream/PPU.vhd
@@ -34,4 +34,5 @@ entity SPPU is
 		PAL			: in std_logic;
 		BLEND			: in std_logic;
+		DIS_SHORTLINE	: in std_logic;
 
 		HIGH_RES		: out std_logic;
@@ -316,5 +317,5 @@ begin
 		if ENABLE = '0' then
 			DOT_CYCLES := "100";
-		elsif V_CNT = 240 and BGINTERLACE = '0' and FIELD = '1' and PAL = '0' then
+		elsif V_CNT = 240 and BGINTERLACE = '0' and FIELD = '1' and PAL = '0' and DIS_SHORTLINE = '0' then
 			DOT_CYCLES := "100";
 		elsif H_CNT = 323 or H_CNT = 327 then
