diff --git a/rtl/upstream/SNES.vhd b/rtl/upstream/SNES.vhd
index d24e180..95dc458 100644
--- a/rtl/upstream/SNES.vhd
+++ b/rtl/upstream/SNES.vhd
@@ -15,7 +15,8 @@ entity SNES is
 		ENABLE		: in std_logic;
 		PAL			: in std_logic;
 		BLEND			: in std_logic;
-		
+		DIS_SHORTLINE	: in std_logic;
+
 		CA       	: out std_logic_vector(23 downto 0);
 		CPURD_N		: out std_logic;
 		CPUWR_N		: out std_logic;
@@ -105,6 +106,7 @@ entity SNES is
 		SS_DI       : in std_logic_vector(7 downto 0);
 		SS_SPC_DO   : out std_logic_vector(7 downto 0);
 		SS_PPU_DO   : out std_logic_vector(7 downto 0);
+		SS_NMI_FORCE: in std_logic := '0';
 
 		TURBO			: in std_logic;
 		
@@ -256,6 +258,7 @@ begin
 
 		TURBO			=> TURBO,
 		SS_BUSY		=> SS_BUSY,
+		SS_NMI_FORCE=> SS_NMI_FORCE,
 		DBG_CPU_EN	=> DBG_CPU_EN
 	);
 
@@ -341,6 +344,7 @@ begin
 		
 		BLEND			=> BLEND,
 		PAL			=> PAL,
+		DIS_SHORTLINE	=> DIS_SHORTLINE,
 		HIGH_RES		=> HIGH_RES,
 		DOTCLK		=> DOTCLK,
 		FIELD_OUT	=> FIELD_OUT,
